### PR TITLE
Rfd36 manifest updates

### PR DESCRIPTION
**Pull request by misterbisson**

### CODE REVIEW

File: rfd/0036/projects/manifest.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  redis:
+    service_type: continuous
+    compute_type: docker
+    image: autopilotpattern/redis:latest
+    package: g4-general-4G
+```
+
+[Please see the service manifest details elsewhere](https://github.com/joyent/rfd/blob/master/rfd/0036/service-manifest.md).
+
+Additional object types that might be expected in future iterations of the project manifest include [network fabrics](https://docs.joyent.com/public-cloud/network/sdn), [firewall rules](https://docs.joyent.com/public-cloud/network/firewall), [RFD36 volumes](https://github.com/joyent/rfd/blob/master/rfd/0026/README.md), and Manta storage "buckets".
```
Comment (misterbisson) [inline rfd/0036/projects/manifest.md]: CNS should be added here.

File: rfd/0036/services/manifest.md [hunk 2]
```diff
@@ -296,22 +386,26 @@ Example:
...
+- `enable`: Default is `notify|notify-only`; can also be `true|1` or `false|0`
+- `min`: The minimum number of instances of the service; default is `1`
+- `max`: The maximum number of instances, but how
+- `retrigger_delay`: The amount of time to wait after a scaling event before aknowledging new events; default is `60s`
+- `notify_email`: one or more email addresses to notify when scaling events are triggered
 
 ```yaml
-  - timeout: -1
-  - preservestopped: always
+autoscale: XXX # Needs more detail
```
Comment (misterbisson) [inline rfd/0036/services/manifest.md:405]: Also need to add details to specify CNS `svc` tag.

File: rfd/0036/projects/triton-cli.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+## `triton project (get|show) <project uuid or name>`
+
+Show the project manifest and details for the specified project.
+
+## `triton project update <project uuid or name> <project manifest> (rolling=<positive int>) (canary|count=<positive int>)`
+
+Add a new project version with the given manifest, set that project version as the default, and trigger a `reprovision` of all service instances to the new version.
+
+Optional arguments passed through to [`reprovision`](#triton-service-reprovisionrestart-service-uuid-or-name-versionversion-uuid-imageimagespectag-instancenameuuid-compute_nodeuuid-countcanarypositive-integer-rollingpositive-integer) (see defaults in `reprovision`):
```
Comment (misterbisson) [inline rfd/0036/projects/triton-cli.md:29]: This made sense in the context of services, but it makes much less sense in the context of a whole project. It's doubtful that the same rules should apply to every service of the project.

File: rfd/0036/services/manifest.md [hunk 4]
```diff
@@ -180,31 +249,40 @@ For `compute_type`:
...
 
-This tells Mariposa how to connect to ContainerPilot's [telemetry interface](https://www.joyent.com/blog/containerpilot-telemetry). Telemetry will eventually expose all ContainerPilot state, including healthchecks and upstreams, in addition to the user-defined performance metrics. The state of instances can be aggregated by the scheduler and used to manage the overall application.
+This tells Mariposa how to connect to ContainerPilot's [telemetry interface](https://www.joyent.com/blog/containerpilot-telemetry). Telemetry will eventually expose all ContainerPilot state, including health checks and upstreams, in addition to the user-defined performance metrics. The state of instances can be aggregated by the scheduler and used to manage the overall application.
+
+Examples:
 
+```yaml
+containerpilot:
+    - network: <network name|uuid> # defaults to the project's default fabric
+    - port: 9090 # defaults to 9090
```
Comment (misterbisson) [inline rfd/0036/services/manifest.md:270]: Also add path?

File: rfd/0036/services/manifest.md [hunk 5]
```diff
@@ -216,6 +294,7 @@ Additional details are required for each type. Appropriate limits need to be ide
 Example:
 
 ```yaml
+healthchecks:
```
Comment (misterbisson) [inline rfd/0036/services/manifest.md:297]: @jasonpincin has done some deeper thinking about health checks. I need to integrate his work here.

**Comment by misterbisson** [comment]
Merged to master in https://github.com/joyent/rfd/commit/c64aaa9dca2c022609c213a05d9d21115911c48d

File: rfd/0036/README.md [hunk 6]
```diff
@@ -15,48 +15,99 @@ state: predraft
...
+          |                     |
+      +---v----+             +--v--+
+      |Projects|<----------->|Users|
+      +--------+             +-----+
+```
 
 ### Service
 
-Services are at the core of Mariposa. A service is defined by a service manifest that specifies a single image (any image supported by IMGAPI) or Manta job command. A service may be run in a single compute instance, or can be scaled scaled to any number of instances as needed.
+Services are at the core of Mariposa. Services are any number of compute instances running the same software image and configuration. A service may be run in a single compute instance, or can be scaled scaled to any number of instances as needed.
```
Comment (richardkiene) [inline rfd/0036/README.md:57]: s/scaled scaled/scaled /g

File: rfd/0036/README.md [hunk 7]
```diff
@@ -15,48 +15,99 @@ state: predraft
...
+- A project may have any number of users
 
-A project is a collection of related compute instances and other related resources ([see "projects" in RFD13](../0013/README.md#proposal) for more detail). A service and all its instances _must_ be a member of a single project. Permissions about who can view or modify a service are set according to RFD13 rules for the project.
+The model for this is GitHub's [organizations, users, and repositories](https://help.github.com/categories/setting-up-and-managing-organizations-and-teams/). Projects (similar to GitHub's repositories) are further defined below and throughout this document.
+
+We will expand the following diagram with additional components as we introduce them:
+
+```
+              +-------------+
+          +---+Organizations| --+
```
Comment (richardkiene) [inline rfd/0036/README.md:47]: The use of `-` and `+` for connecting lines are inconsistent here and elsewhere.

File: rfd/0036/README.md [hunk 8]
```diff
@@ -67,23 +118,141 @@ These types may include:
...
-- `kvm|vm|hvm`
-- `manta`
 
-Only `docker`,`infrastructure`, and `kvm` are required for the MVP.
+### Project
 
-### Task queue
+While [services](#service) abstract any number of compute instances running the same software with the same configuration, a project allows users to group services and other resources together so that they can be managed as a whole without the distraction or complication or unrelated components.
+
+The concept of projects was first introduced with [RBACv2 in RFD13](../0013/README.md#proposal), this RFD intends to replace the definition of services from RFD13. Though projects and other features described in this RFD may be built independently of RBACv2 work, many assumptions about RBACv2 are made throughout this text.
```
Comment (richardkiene) [inline rfd/0036/README.md:128]: If you intend to replace the RFD13 definition of services, is that replacement congruent with the SECAPI work that Alex is doing?

File: rfd/0036/README.md [hunk 9]
```diff
@@ -67,23 +118,141 @@ These types may include:
...
+    |
++---v---+
+|Compute|
++-------+
+```
+
+In the above diagram "unmanaged compute" describes both existing instances that were defined before the introduction of services, as well as new instances that a user may define without first defining a service. Support for existing unmanaged compute and their ongoing use, as well as the ability to provision new unmanaged instances is required, despite the introduction of services. However, there is no requirement nor intention of providing a migration plan to convert a collection of existing unmanaged instances into a service.
 
-Scaling, upgrading, even stopping all the instances of an app can take time...sometimes significant time. To represent this to the user, Triton must expose the task queue and offer the ability to cancel jobs.
+Projects are described by a [project manifest](./projects/manifest.md), a YAML-formatted file that can be easily copied from elsewhere and in which changes are easily discernible in a text diff. They also have attached metadata as described below.
```
Comment (richardkiene) [inline rfd/0036/README.md:156]: YAML? Ugh...

File: rfd/0036/README.md [hunk 10]
```diff
@@ -67,23 +118,141 @@ These types may include:
...
 
-This document defines tasks specific to deploying services, but the task queue is not limited to services.
 
 ### Project meta (including secrets)
 
 Many applications require configuration values which are undesirable or unsafe to set in the application image. These can include license keys, a flag setting whether it's a staging or production environment, usernames and passwords, and other details.
 
 This document proposes a simple method of storing those details and injecting them into containers. It is not intended to provide the rich features of solutions like Hashicorp's Vault, instead it is intended to provide a basic solution that is easy to use in a broad variety of applications.
+
+[Read more about meta in the Mariposa context](./meta), including Triton CLI commands and the manifest file.
```
Comment (richardkiene) [inline rfd/0036/README.md:167]: >It is not intended to provide the rich features of solutions like Hashicorp's Vault, instead it is intended to provide a basic solution that is easy to use in a broad variety of applications.

I'm not sure it is appropriate to hand wave over the details here. From my perspective this reads a lot like "Vault is difficult, ours will work, but it's not all that secure".

File: rfd/0036/README.md [hunk 11]
```diff
@@ -67,23 +118,141 @@ These types may include:
...
+- [Creating, copying, and moving projects like microsite.jupiter.example.com](./user-stories/microsite-jupiter-example-com.md) (also includes secret management)
+
+
+## Implementation and architecture
+
+This RFD is mostly concerned with what users can do with Mariposa, not how those features are implemented. Work to define an implementation has begun in additional RFDs, which propose the following components:
+
+- [Projects API](https://github.com/joyent/rfd/blob/master/rfd/0079/README.md), which is where [projects](https://github.com/joyent/rfd/blob/master/rfd/0036/project.md) and their attached [services](https://github.com/joyent/rfd/blob/master/rfd/0036/service.md) and [metadata](https://github.com/joyent/rfd/blob/master/rfd/0036/meta.md) are managed
+- [ProjectsConvergence API](https://github.com/joyent/rfd/blob/master/rfd/0080/README.md), which is responsible for watching the actual state of the project and its services, comparing that to the goal state, developing a plan to reconcile those differences, and maintaining a queue of reconciliation plans currently being executed.
```
Comment (richardkiene) [inline rfd/0036/README.md:193]: I'm not sure I like `ProjectsConvergence` as the name here. `ProjectsWatcher` perhaps?

File: rfd/0036/README.md (hunk 8 above)
Comment (misterbisson) [reply rfd/0036/README.md:128]: The requirements for what's implemented in projects and services are being defined here. RFD48 will need to be reviewed to ensure it offers the features to support usage described herein.

File: rfd/0036/README.md (hunk 9 above)
Comment (misterbisson) [reply rfd/0036/README.md:156]: You have an alternative that supports inline comments?

File: rfd/0036/README.md (hunk 11 above)
Comment (misterbisson) [reply rfd/0036/README.md:193]: "Convergence" is the term of art in this space. The emphasis is on reconciling differences...the work it does is to create plans to converge the actual state with the goal state. The work of watching for changes is done by other services and reported back to this service.

**Pull request closed without merging**